azure/Kqlmagic/adal_token_cache.py

```diff
@@ -117,14 +148,45 @@ def _query_cache(self, is_mrrt, user_id, client_id):
                 matches.append(v)
         return matches
 
+    def clear_db(self):
+
+        items = self.db.items()
+
+        for token_tuple in items:
+
+            db_key = token_tuple[0] # 'kqlmagicstore/tokens/USER_ID' address
+            data = token_tuple[1] #encrypted token
+            time_created, _unused = Fernet._get_unverified_token_data(data)  #get token timestamp 
+            EXP_TIME = 1000 #in seconds
+            if(time_created <time.time()- EXP_TIME):
+                del self.db[db_key]
+
     def _save(self):
+        ####
+        self.clear_db()
+        #####
         data = self.serialize()
-        self.db['kqlmagicstore/tokens'] = data
+        data_as_bytes = data.encode()
+
+        #init fernet
+        fernet = self.fernet
+
+        #encrypt 
+        data_encrypted = fernet.encrypt(data_as_bytes)
+        
+        self.db['kqlmagicstore/tokens/'+ str(self.user_id)] = data_encrypted
 
     def _restore(self):
         try:
-            data = self.db['kqlmagicstore/tokens']
+            data_encrypted = self.db['kqlmagicstore/tokens/'+ str(self.user_id)]
+            fernet = self.fernet
+
+            #decrypt
+            data_decrypted_as_bytes = fernet.decrypt(data_encrypted)
+            data = data_decrypted_as_bytes.decode()
         except KeyError:
-            print("no stored tokens")
-        else:
-            self.deserialize(data)
+            # print("no stored tokens")
```

Comment:
the logic that there is no stored token, shoudn't be using an exception flow, you could check that the data_encrypted is None